### PR TITLE
[feature] Interactions v2

### PR DESCRIPTION
**Pull request by l7ssha**
- MessageBuilder only way to send messages (2075c75)
- Implement lightweight version of TextChannel which implements Cacheable which allows performing basic channel operation (https://github.com/l7ssha/nyxx/pull/115/commits/42082b114fa98dbc575261d7398bf496c5ff9fa0)
- Integrate buttons into nyxx (read only). Change naming to component*. (02a1d5a)

### CODE REVIEW

File: nyxx/lib/src/utils/BuilderUtility.dart [hunk 1]
```diff
@@ -9,4 +9,8 @@ class BuilderUtility {
   /// Build the data from [AllowedMentions]
   static Map<String, dynamic> buildRawAllowedMentions(AllowedMentions mentions) =>
       mentions._build();
+
+  static Map<String, dynamic> build<T extends Builder> (T builder) => builder._build();
+
+  static Map<String, dynamic> buildWithClient<T extends BuilderWithClient> (T builder, INyxx client) => builder.build(client);
```
Comment (l7ssha) [inline nyxx/lib/src/utils/BuilderUtility.dart:15]: Need docs

Comment (HarryET) [reply nyxx/lib/src/utils/BuilderUtility.dart:15]: docs still required.

File: nyxx_commander/lib/src/utils.dart [hunk 2]
```diff
@@ -1,11 +1,10 @@
 part of nyxx_commander;
 
-// TODO: FIX. I think that is just awful but is does its job
-extension _CommandMatcher on Iterable<CommandEntity> {
-  CommandEntity? _findMatchingCommand(Iterable<String> messageParts) {
-    for (final entity in this) {
-      if (entity is CommandGroup && entity.name == "") {
-        final e = entity._commandEntities._findMatchingCommand(messageParts);
+class  _CommandMatcher {
+  static CommandEntity? _findMatchingCommand(Iterable<String> messageParts, Iterable<CommandEntity> commands) {
```
Comment (HarryET) [inline nyxx_commander/lib/src/utils.dart:4]: Does this need documentations? Possibly for contributors? same for class line above.

File: nyxx_interactions/lib/src/builders/SlashCommandBuilder.dart [hunk 3]
```diff
@@ -14,6 +14,7 @@ class SlashCommandBuilder implements Builder {
   /// The arguments that the command takes
   List<CommandOptionBuilder> options;
 
+  // Handler for SlashCommandBuilder
```
Comment (HarryET) [inline nyxx_interactions/lib/src/builders/SlashCommandBuilder.dart:17]: using `//` insted of `///`

**Review by HarryET — APPROVED** [review]

**Pull request merged**